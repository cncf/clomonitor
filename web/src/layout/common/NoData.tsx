import styles from './NoData.module.css';

interface Props {
  children: string | JSX.Element | JSX.Element[];
  className?: string;
}

const NoData = (props: Props) => (
  <div
    role="alert"
    className={`alert ms-auto me-auto rounded-0 my-5 text-center p-4 p-sm-5 border border-primary ${styles.wrapper} ${props.className}`}
  >
    <div className="h4">{props.children}</div>
  </div>
);

export default NoData;
