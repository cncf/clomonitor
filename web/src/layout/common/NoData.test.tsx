import { render, screen } from '@testing-library/react';

import NoData from './NoData';

describe('NoData', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(
      <NoData>
        <>no data</>
      </NoData>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(
      <NoData>
        <>no data</>
      </NoData>
    );

    expect(screen.getByRole('alert')).toBeInTheDocument();
    expect(screen.getByRole('alert')).toHaveTextContent('no data');
  });
});
