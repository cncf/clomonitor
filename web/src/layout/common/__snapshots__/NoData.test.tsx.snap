// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoData creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert ms-auto me-auto rounded-0 my-3 my-md-5 text-center p-4 p-sm-5 border border-primary wrapper undefined"
    role="alert"
  >
    <div
      class="h4"
    >
      no data
    </div>
  </div>
</DocumentFragment>
`;
